default/methods/source_control/file/file_browser.py

```diff
@@ -575,55 +612,58 @@ def __build_media_type_value(self) -> str:
+            return None, None
 
-        # For creating / viewing Jobs
+        return working_dir_file_list, count
+    
+    def get_job_file_list(self):
+        
+        # TODO permissions check on job id?
 
-        if self.metadata['file_view_mode'] == "task" and self.metadata['job_id']:
+        # TODO handling for larger file sizes
 
-            # TODO permissions check on job id?
+        # TODO better null handling here... a lot of assumptions
+        # Would prefer to declare "new" job condition
+        # Instead of inferring from file_view_mode and job_id being present
 
-            # TODO handling for larger file sizes
+        job = Job.get_by_id(session = self.session,
+                            job_id = self.metadata['job_id'])
 
-            # TODO better null handling here... a lot of assumptions
-            # Would prefer to declare "new" job condition
-            # Instead of inferring from file_view_mode and job_id being present
+        file_list_attached_to_job = WorkingDirFileLink.file_list(
+            session = self.session,
+            working_dir_id = job.directory_id,
+            limit = None)
 
-            job = Job.get_by_id(session = self.session,
-                                job_id = self.metadata['job_id'])
+        # TODO future maybe just get ids only from sql
+        ignore_id_list = [i.id for i in file_list_attached_to_job]
+        output_job_file_list = []
 
-            file_list_attached_to_job = WorkingDirFileLink.file_list(
-                session = self.session,
-                working_dir_id = job.directory_id,
-                limit = None)
+        for index_file_attach, file in enumerate(file_list_attached_to_job):
+            file_serialized = file.serialize_with_type(session = self.session)
+            output_job_file_list.append(file_serialized)
+            output_job_file_list[index_file_attach]['attached_to_job'] = True
 
-            # TODO future maybe just get ids only from sql
-            ignore_id_list = [i.id for i in file_list_attached_to_job]
+        return output_job_file_list, ignore_id_list
 
-            for index_file_attach, file in enumerate(file_list_attached_to_job):
-                file_serialized = file.serialize_with_type(session = self.session)
-                output_file_list.append(file_serialized)
-                output_file_list[index_file_attach]['attached_to_job'] = True
 
+    def get_all_file_list(self, ignore_id_list = None):
+        policy_engine = PolicyEngine(session = self.session, project = self.project)
+        perm_result = policy_engine.member_has_perm(
+            member = self.member,
+            object_id = self.directory.id,
+            object_type = ValidObjectTypes.dataset,
+            perm = DatasetPermissions.dataset_view
+        )
+        if not perm_result.allowed:
+            self.log['error']['unauthorized'] = f'Cannot view dataset ID: {self.directory.id}'
+            return None, None
```

Comment:
Previously, the following error handling would occur in this case:

```
resp = jsonify(log=self.log)
resp.status = 401
raise Unauthorized(response = resp)
```

now instead we just write to the log, since it allows us to pull out the error handling [here](https://github.com/diffgram/diffgram/pull/1536/files#diff-84ca682673d0f6d96b6c08959622d617aca63ab090e97754044e22cc6d5b717fR620). I tested the behaviour of both from the UI and found there was no difference, but let me know if its important to still raise an exception here